src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -14,44 +14,96 @@
 USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0'
-GENOME_STATUS_FILENAME = "genome_status.csv"
+GENOME_STATUS_FILENAME = "genome_status.json"
 
 
 class GenomeStatus:
-    """To represent the data status for a certain genome ID.
+    """A class to represent the status of a single genome.
 
-    Attributes:
-        original_id(str): original ID for the genome.
-        resolved_refseq_id(str): RefSeq ID generated by resolving the original genome ID.
-        attempted(str): If the attempt of resolving the genome ID has already been done. Defaults to `False`.
-        filename(str): Path to the file that stores the genome data. Defaults to "".
+    The status of genomes is tracked in a JSON file which has a name defined
+    in variable `GENOME_STATUS_FILENAME`.
     """
 
     def __init__(self,
                  original_id: str,
-                 resolved_refseq_id: str,
-                 attempted: str = 'False',
-                 filename: str = ""):
+                 resolved_refseq_id: str = "",
+                 resolve_attempted: bool = False,
+                 bgc_path: str = ""):
+        """Initialize a GenomeStatus object for the given genome.
+
+        Args:
+            original_id (str): The original ID of the genome.
+            resolved_refseq_id (str, optional): The resolved RefSeq ID of the
+                genome. Defaults to "".
+            resolve_attempted (bool, optional): A flag indicating whether an
+                attempt to resolve the RefSeq ID has been made. Defaults to False.
+            bgc_path (str, optional): The path to the downloaded BGC file for
+                the genome. Defaults to "".
+        """
         self.original_id = original_id
-        self.resolved_refseq_id = "" if resolved_refseq_id == 'None' else resolved_refseq_id
-        self.attempted = attempted == 'True'
-        self.filename = filename
+        self.resolved_refseq_id = resolved_refseq_id
+        self.resolve_attempted = resolve_attempted
+        self.bgc_path = bgc_path
+
+    @staticmethod
+    def load_from_json(file: str | PathLike) -> dict[str, 'GenomeStatus']:
+        """Get a dict of GenomeStatus objects by loading given genome status file.
 
-    def to_csv(self, file: str | PathLike, mode: str = 'a') -> None:
-        """Write object to a csv file.
+        Note that an empty dict is returned if the given file doesn't exist.
+
+        Args:
+            file(str | PathLike): Path to genome status file.
+
+        Returns:
+            dict: dict keys are genome original id and values are GenomeStatus
+                objects. An empty dict is returned if the given file doesn't exist.
+        """
+        genome_status_dict = {}
+        if Path(file).exists():
+            with open(file, "r") as f:
+                data = json.load(f)
+            genome_status_dict = {
+                gs["original_id"]: GenomeStatus(**gs)
```

Comment:
Just for me to know (couldn't find a satisfying answer online), what is the usage of the double star operator for initializing a class? Just assigning the values of each key to the parameters of the init function, matching the keys with the init parameters?

Reply:
it's called keyword argument, see the doc https://docs.python.org/3.10/glossary.html#term-argument